tests/e2e-server/src/test/java/org/glassfish/jersey/tests/e2e/server/OptionalParamConverterNoProviderTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.jersey.tests.e2e.server;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Response;
+
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.Test;
+
+public class OptionalParamConverterNoProviderTest extends JerseyTest {
+
+    private static final String PARAM_NAME = "paramName";
+
+    @Path("/OptionalResource")
+    public static class OptionalResource {
+
+        @GET
+        @Path("/fromString")
+        public Response fromString(@QueryParam(PARAM_NAME) Optional<String> data) {
+            return Response.ok(data.orElse("")).build();
+        }
+
+        @GET
+        @Path("/fromInteger")
+        public Response fromInteger(@QueryParam(PARAM_NAME) Optional<Integer> data) {
+            return Response.ok(data.orElse(0)).build();
+        }
+
+        @GET
+        @Path("/fromList")
+        public Response fromList(@QueryParam(PARAM_NAME) List<Optional<Integer>> data) {
+            StringBuilder builder = new StringBuilder("");
+            for (Optional<Integer> val : data) {
+                builder.append(val.orElse(0));
+            }
+            return Response.ok(builder.toString()).build();
+        }
```

Comment:
Would it make sense to have a corresponding test for `List<OptionalInt>>`, `List<OptionalLong>>`, and `List<OptionalDouble>>`?

Reply:
Done